src/nvim/os/unix_defs.h

```diff
@@ -1,7 +1,101 @@
 #ifndef NVIM_OS_UNIX_DEFS_H
 #define NVIM_OS_UNIX_DEFS_H
 
+#include <unistd.h>
+#include <signal.h>
+
+// Defines BSD, if it's a BSD system.
+#ifdef HAVE_SYS_PARAM_H
+# include <sys/param.h>
+#endif
+
+#if defined(DIRSIZ) && !defined(MAXNAMLEN)
+# define MAXNAMLEN DIRSIZ
+#endif
+
+#if defined(NAME_MAX) && !defined(MAXNAMLEN)
+# define MAXNAMLEN NAME_MAX
+#endif
+
+// Default value.
+#ifndef MAXNAMLEN
+# define MAXNAMLEN 512
+#endif
+
+#define BASENAMELEN (MAXNAMLEN - 5)
```

Comment:
Vim's `os_win32.h` defines this as `(_MAX_PATH - 5)` which is 255. Does it matter? Why/why not?